js/MainContent.js

```diff
@@ -305,8 +306,31 @@ function validateUNHCR($elem, throwErrorFlag) {
         return num.replace(/[^ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789]/g, "");
     }
 
-    // function checks if format is valid for the new UNHCR ID style
     function checkValidNew(num) {
+        var format = "^[0-9]{3}-[0-9]{2}-[0-9]{5}$";
+
+        // remove non-alphanumeric values from num
+        num = removeNonAlphanumeric(num);
+
+        // convert letters to uppercase and replace capital O with zeros (0)
+        num = num.replace(/O/g, "0");
+
+        // add '-' in 3rd position and !
+        num = num.substr(0, 3) + "-" + num.substr(3);
+
+        // add '-' in 6th position!
+        num = num.substr(0, 6) + "-" + num.substr(6);
+
+        // if entered data doesn't match format, return false.
+        if (num.match(format) == null) {
+            return false;
+        } else {
+            placeInputValue($elem, num);
+            return true;
+        }
+    }
+    // function checks if format is valid for the new UNHCR ID style
+    function checkValidOld(num) {
         var format = "^[0-9]{3}-[0-9]{2}C[0-9]{5}$";
```

Comment:
I believe you can make one tiny tweak instead of copy / pasting most of this function with one tiny Regex difference. If you change this regex to: `^[0-9]{3}-[0-9]{2}(C|-)[0-9]{5}$` you should be good to go :D

Here's a useful website for testing regex and it even explains what a capture group like this `(C|-)` does. Basically, it matches EITHER `C` OR `-` in that spot, which is exactly what you want right?

You can even make this a [non-capture group](https://stackoverflow.com/questions/3512471/what-is-a-non-capturing-group-in-regular-expressions) like this: `(?:C|-)`, which may be slightly better